File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/ride/UpdateRideUseCase.java
package com.lattis.ellipse.domain.interactor.ride;

import com.lattis.ellipse.data.network.model.response.ride.UpdateTripResponse;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.RideRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 3/30/17.
 */

public class UpdateRideUseCase extends UseCase<UpdateTripResponse> {

    private RideRepository rideRepository;

    private int trip_id;
    private double[][] steps;


    @Inject
    public UpdateRideUseCase(ThreadExecutor threadExecutor,
                             PostExecutionThread postExecutionThread,
                             RideRepository rideRepository) {
        super(threadExecutor, postExecutionThread);
        this.rideRepository = rideRepository;
    }

    public UpdateRideUseCase withTripId(int trip_id) {
        this.trip_id = trip_id;
        return this;
    }

    public UpdateRideUseCase withSteps(double[][] steps) {
        this.steps = steps;
        return this;
    }


    @Override
    protected Observable<UpdateTripResponse> buildUseCaseObservable() {
        return this.rideRepository.updateRide(trip_id, steps);
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/activetripservice/UpdateRideUseCase.kt
package com.lattis.domain.usecase.ride

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.models.UpdateTripData
import com.lattis.domain.repository.RideRepository
import com.lattis.domain.usecase.base.UseCase
import javax.inject.Inject

import io.reactivex.rxjava3.core.Observable


class UpdateRideUseCase @Inject
constructor(threadExecutor: ThreadExecutor,
            postExecutionThread: PostExecutionThread,
            private val rideRepository: RideRepository
) : UseCase<UpdateTripData>(threadExecutor, postExecutionThread) {

    private var trip_id: Int = 0
    private lateinit var steps: Array<DoubleArray>

    fun withTripId(trip_id: Int): UpdateRideUseCase {
        this.trip_id = trip_id
        return this
    }

    fun withSteps(steps: Array<DoubleArray>): UpdateRideUseCase {
        this.steps = steps
        return this
    }


    override fun buildUseCaseObservable(): Observable<UpdateTripData> {
        return this.rideRepository.updateRide(trip_id, steps)
    }
}

